src/max.ts

```diff
@@ -0,0 +1,66 @@
+import ReturnValueType from "./types/ReturnValueType";
+import { isAsyncIterable, isIterable } from "./_internal/utils";
+
+function sync(iterable: Iterable<number>) {
+  let n = NaN;
+  for (const a of iterable) {
+    if (Number.isNaN(a)) {
+      return a;
+    } else if (Number.isNaN(n)) {
+      n = a;
+    } else if (a > n) {
+      n = a;
+    }
+  }
+
+  if (Number.isNaN(n)) {
+    return 0;
+  }
+
+  return n;
+}
+
+async function async(iterable: AsyncIterable<number>) {
+  let n = NaN;
+  for await (const a of iterable) {
+    if (Number.isNaN(a)) {
+      return a;
+    } else if (Number.isNaN(n)) {
+      n = a;
+    } else if (a > n) {
+      n = a;
+    }
+  }
+
+  if (Number.isNaN(n)) {
+    return 0;
+  }
```

Comment:
`Math.max()` returns `-Infinity`. Therefore, it seems appropriate that this code also returns `-Infinity`.